extensions/ql-vscode/test/factories/authentication.ts

```diff
@@ -0,0 +1,38 @@
+import { retry } from "@octokit/plugin-retry";
+import * as Octokit from "@octokit/rest";
+import { RequestInterface } from "@octokit/types/dist-types/RequestInterface";
+
+import { Credentials } from "../../src/common/authentication";
+
+function makeTestOctokit(octokit: Octokit.Octokit): Credentials {
+  return {
+    getOctokit: async () => octokit,
+    getAccessToken: async () => {
+      throw new Error("getAccessToken not supported by test credentials");
+    },
+    getExistingAccessToken: async () => {
+      throw new Error(
+        "getExistingAccessToken not supported by test credentials",
+      );
+    },
+  };
+}
+
+export function testCredentialsWithStub(
+  requestSpy?: jest.SpyInstance<RequestInterface<object>>,
+): Credentials {
+  const request =
+    requestSpy ??
+    jest.fn(async () => {
+      throw new Error("Tried to make HTTP request but no stub was provided");
+    });
+  return makeTestOctokit({ request } as any);
+}
+
+export function testCredentialsWithUnauthenticatedOctokit(): Credentials {
+  return makeTestOctokit(new Octokit.Octokit({ retry }));
+}
```

Comment:
Can we get rid of this one? It's used in one place and the test is using a scenario so responses should be mocked.

Reply:
I've combined the two methods together to just `testCredentialsWithRealOctokit` which now takes an _optional_ `token` parameter. I think that's a little cleaner than it was before.